### PR TITLE
backend test v1

### PR DESCRIPTION
**Pull request by JojoFlower**

### CODE REVIEW

File: backend/v1/generate.sh [hunk 1]
```diff
@@ -0,0 +1,2 @@
+protoc --go_out=product --go-grpc_out=product product/rpc/product.proto
+protoc --go_out=order --go-grpc_out=order order/rpc/order.proto
```
[the remark this answers is not in the document]
Comment (JojoFlower) [reply backend/v1/generate.sh:2]: looks like it's not that hard, i just add it as a comment 😛 
image

File: backend/v1/order/rpc/order.proto [hunk 2]
```diff
@@ -0,0 +1,19 @@
+syntax = "proto3";
+
+package order;
+
+option go_package = "rpc;order";
```
[the remark this answers is not in the document]
Comment (JojoFlower) [reply backend/v1/order/rpc/order.proto]: actually, it looks like it generates inside the project

File: backend/v1/README.md [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+<li>address line (45 Rue des Petites Ecuries)</li>
+<li>postal code (75010)</li>
+<li>city (Paris)</li>
+<li>country (FR)</li>
+</ul>
+</ol>
+<h4>II- Address validation</h4>
+As discussed previously, you should propose and implement a solution to validate the shipping address of an order before creating it:
+<ul>
+<li>If the address contains some slight errors and the correct data can be identified with certainty by the system, the address will be automatically fixed (e.g. "Aubervillrs" → "Aubervilliers").
```
[the remark this answers is not in the document]
Comment (JojoFlower) [reply backend/v1/README.md]: i will ajust the examples

File: backend/v1/order/rpc/order.proto (hunk 2 above)
[the remark this answers is not in the document]
Comment (CristianPalmaBigBlue) [reply backend/v1/order/rpc/order.proto]: The error when using rpc without the slash said also: The import path must contain at least one forward slash ('/') character.
And, indeed, it worked as expected generating the proto code inside the rpc folder.

File: backend/v1/order/rpc/order.pb.go [hunk 4]
```diff
@@ -0,0 +1,270 @@
+// Code generated by protoc-gen-go. DO NOT EDIT.
```
[the remark this answers is not in the document]
Comment (CristianPalmaBigBlue) [reply backend/v1/order/rpc/order.pb.go:1]: As during the test the candidate should update the .proto file, they should regenerate those files anyway. Having those files does not help the candidate.

File: backend/v1/README.md [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+<li>Introduction to gRPC: <a href="https://grpc.io/docs/what-is-grpc/introduction/">https://grpc.io/docs/what-is-grpc/introduction</a></li>
+<li>gRPC with Golang: <a href="https://grpc.io/docs/languages/go/">https://grpc.io/docs/languages/go</a></li>
+<li>Protocol Buffers: <a href="https://developers.google.com/protocol-buffers/docs/proto3">https://developers.google.com/protocol-buffers/docs/proto3</a></li>
+</ul>
+<h4>Setup</h4>
+<ol>
+<li>Install Golang: <a href="https://golang.org/doc/install#download">https://golang.org/doc/install#download</a></li>
+<li>Install Protoc to perform code generation: <a href="./doc/protoc.md">protoc</a></li>
+<li>Install the Go packages of the project: run <code>go install</code> from <code>go.mod</code> directory level</li>
+<li>When editing proto files, code generation can be triggered by running <code>generate.sh</code></li>
```
[the remark this answers is not in the document]
Comment (CristianPalmaBigBlue) [reply backend/v1/README.md]: To do not have to update manually the latest version, `@latest` is a good choice.
```
go install google.golang.org/protobuf/cmd/protoc-gen-go@latest
go install google.golang.org/grpc/cmd/protoc-gen-go-grpc@latest
```

**Pull request merged**